src/braket/circuits/circuit.py

```diff
@@ -440,22 +441,38 @@ def _check_if_qubit_measured(
         Raises:
             ValueError: If adding a gate or noise operation after a measure instruction.
         """
-        if (
-            # check if there is a measure instruction on the target qubit
-            target
-            and target in self._measure_targets
-            # check if there is a measure instruction on any qubits in the target_mapping
-            or (target_mapping and any(targ in self._measure_targets for targ in target_mapping))
-            # If no target or target_mapping is supplied, check if there is a measure
-            # instruction on the current instructions target qubit
-            or (
-                instruction.target
-                and any(targ in self._measure_targets for targ in instruction.target)
+        if self._measure_targets:
+            measure_on_target = (
+                target and any(tar in self._measure_targets for tar in target)
+                if isinstance(target, Iterable)
+                else target in self._measure_targets
             )
-        ):
-            raise ValueError(
-                "cannot add a gate or noise operation on a qubit after a measure instruction."
+            measure_on_target_mapping = target_mapping and any(
+                targ in self._measure_targets for targ in target_mapping.values()
             )
+            measure_on_instruction_target = (
+                instruction
+                and instruction.target
+                and any(targ in self._measure_targets for targ in instruction.target)
+            )
+            if (
+                # check if there is a measure instruction on the targeted qubit(s)
+                not isinstance(instruction.operator, Measure)
+                and (
+                    measure_on_target or measure_on_target_mapping or measure_on_instruction_target
+                )
+            ):
+                raise ValueError(
+                    "cannot add a gate or noise operation on a qubit after a measure instruction."
+                )
+
+            # Check if the target_qubits are already measured
+            if isinstance(instruction.operator, Measure) and measure_on_instruction_target:
+                intersection = set(instruction.target) & set(self._measure_targets)
+                raise ValueError(
+                    f"cannot measure the same qubit(s) {', '.join(map(str, intersection))} "
+                    "more than once."
+                )
```

Comment:
Can this be combine as well, and with the error message "Cannot apply instruction to measured qubits"

Reply:
Yes, it can. I will combine these two errors